From 2e669e04574d319804fd4393c173611d998308ee Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sun, 2 May 2021 09:53:41 +0200
Subject: [PATCH] Add AppStream metadata file

It's useful for software catalogs to display detailed information about this application.
---
 Makefile                  |  5 ++++-
 data/griffith.appdata.xml | 28 ++++++++++++++++++++++++++++
 2 files changed, 32 insertions(+), 1 deletion(-)
 create mode 100644 data/griffith.appdata.xml

diff --git a/Makefile b/Makefile
index afb4f9b..b511c68 100755
--- a/Makefile
+++ b/Makefile
@@ -33,6 +33,7 @@ IMPORTPLUGINSDIR = $(PLUGINSDIR)/imp
 EXTENSIONSPLUGINSDIR = $(PLUGINSDIR)/extensions
 TPLDIR = $(DATADIR)/export_templates
 APPLICATIONSDIR = $(PREFIX)/share/applications
+METAINFODIR = $(PREFIX)/share/metainfo
 ICONDIR = $(PREFIX)/share/icons/hicolor/scalable/apps
 LOCALEDIR = $(PREFIX)/share/locale
 BASHCOMPDIR = $(PREFIX)/share/bash-completion/completions
@@ -54,7 +55,7 @@ install:
 	@echo "^^^^^^^^^^^^^^^^^^^"
 	$(INSTALL) -m 755 -d $(BINDIR) $(DATADIR) $(LIBDIR) $(LIBDIR)/db \
 		$(PLUGINSDIR) $(MOVIEPLUGINSDIR) $(EXPORTPLUGINSDIR) $(IMPORTPLUGINSDIR) $(EXTENSIONSPLUGINSDIR) \
-		$(FONTSDIR) $(APPLICATIONSDIR) $(ICONDIR) $(TPLDIR) \
+		$(FONTSDIR) $(APPLICATIONSDIR) $(METAINFODIR) $(ICONDIR) $(TPLDIR) \
 		$(IMAGESDIR) $(GLADEDIR) $(BASHCOMPDIR)
 	$(INSTALL) -m 755 griffith $(LIBDIR)
 	$(INSTALL) -m 644 lib/*.py $(LIBDIR)
@@ -69,6 +70,7 @@ install:
 	$(INSTALL) -m 644 images/*.png $(IMAGESDIR)
 	$(INSTALL) -m 644 images/griffith.svg $(ICONDIR)
 	$(INSTALL) -m 644 data/griffith.desktop $(APPLICATIONSDIR)
+	$(INSTALL) -m 644 data/griffith.appdata.xml $(METAINFODIR)
 	$(INSTALL) -m 644 data/bash_completion $(BASHCOMPDIR)/griffith
 	
 	# installing language files
@@ -106,6 +108,7 @@ uninstall:
 	${RM} -r $(DATADIR)
 	${RM} -r $(ICONDIR)/griffith.svg
 	${RM} -r $(APPLICATIONSDIR)/griffith.desktop
+	${RM} -r $(METAINFODIR)/griffith.appdata.xml
 	for lang in $(LANGUAGES); do \
 		${RM} -r $(LOCALEDIR)/$$lang/LC_MESSAGES/griffith.mo; \
 	done
diff --git a/data/griffith.appdata.xml b/data/griffith.appdata.xml
new file mode 100644
index 0000000..c7a696c
--- /dev/null
+++ b/data/griffith.appdata.xml
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>griffith.desktop</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-2.0</project_license>
+  <name>Griffith</name>
+  <summary>Movie collection manager</summary>
+  <description>
+    <p>Griffith is a film collection manager with ease of use in mind.</p>
+    <ul>
+      <li>Fetch film information from the web using a plugin system</li>
+      <li>Automatic numbering</li>
+      <li>Backup/restore functions</li>
+      <li>Exports listings to PDF and CSV files</li>
+      <li>Generates a PDF with film's cover</li>
+      <li>Clone film information</li>
+      <li>Loans management</li>
+      <li>View filtering</li>
+    </ul>
+  </description>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://screenshots.debian.net/shrine/screenshot/22/simage/large-0cde3999904a6899300369f586e363dc.png</image>
+    </screenshot>
+  </screenshots>
+  <url type="homepage">https://gitlab.com/Strit/griffith</url>
+  <url type="bugtracker">https://gitlab.com/Strit/griffith/issues</url>
+</component>
-- 
2.31.1

